File: 908-middle-of-the-linked-list/middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {

        ArrayList<ListNode> list = new ArrayList<>();

         ListNode curr = head;

        while(curr!=null){
            list.add(curr);
            curr=curr.next;
        }

        int mid = list.size()/2;

        return list.get(mid);
        
    }
}

File: 908-middle-of-the-linked-list/middle-of-the-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {

    // 1. Convert a Linkedlist to an array

    let arr = [];

    let curr = head;

    while(curr){
        arr.push(curr);
        curr = curr.next;
    }

    // To find middle node of the list

    let mid = Math.floor(arr.length/2);

    return arr[mid]; 
    
};